guides/source/asset_pipeline.md

```diff
@@ -8,479 +8,616 @@ This guide covers the asset pipeline.
-already available in `public/assets` as a fingerprinted file, then that path is
-referenced.
-
-If you want to use a [data URI](https://en.wikipedia.org/wiki/Data_URI_scheme) -
-a method of embedding the image data directly into the CSS file - you can use
-the `asset_data_uri` helper.
-
-```css
-#logo { background: url(<%= asset_data_uri 'logo.png' %>) }
-```
+WARNING: If you're precompiling your assets (see [the Production
+section](#production) below), linking to an asset that doesn't exist will
+raise an exception in the calling page. This includes linking to a blank string.
+Be careful when using `image_tag` and other helpers with user-supplied data.
 
-This inserts a correctly-formatted data URI into the CSS source.
+Development vs. Production
+--------------------------
 
-Note that the closing tag cannot be of the style `-%>`.
+### Development
 
-### Raise an Error When an Asset is Not Found
+In development, Rails and Propshaft are configured for flexibility, allowing
+rapid iteration without manual intervention. This setup makes it easier to
+manage assets as they change frequently during development.
 
-If you are using sprockets-rails >= 3.2.0 you can configure what happens
-when an asset lookup is performed and nothing is found. If you turn off "asset fallback"
-then an error will be raised when an asset cannot be found.
+#### No Caching in Development
 
-```ruby
-config.assets.unknown_asset_fallback = false
-```
+In development, Rails is configured to bypass asset caching. This means that
+when you modify assets (e.g., CSS, JavaScript), Rails will serve the most
+up-to-date version directly from the file system. There's no need to worry about
+versioning or file renaming because caching is skipped entirely. Browsers will
+automatically pull in the latest version each time you reload the page.
 
-If "asset fallback" is enabled then when an asset cannot be found the path will be
-output instead and no error raised. The asset fallback behavior is disabled by default.
+#### Automatic Reloading of Assets
 
-### Turning Digests Off
+Rails ensures that asset changes are immediately reflected in the browser. It
+checks for updates with every request, so that when you edit an asset like a
+JavaScript or CSS file, reloading the page will display the changes instantly
+without needing to restart the server.
 
-You can turn off digests by updating `config/environments/development.rb` to
-include:
+#### Propshaft Integration in Development
 
-```ruby
-config.assets.digest = false
-```
+The integration of Propshaft in a Rails application is handled by the standard
+Rails development environment. Propshaft will automatically serve and manage
+assets by compiling and serving updated versions directly from the file system
+without manual intervention. This is done as part of the usual Rails asset
+pipeline, with the default configuration ensuring the most recent asset changes
+are served on page reload.
```

Comment:
This doesn't seem to say anything new, feels repetitive..

Reply:
I removed this section completely and moved some of the unique information into the other sections.